packages/framework/client-logger/fluid-telemetry/src/test/containerTelemetry.spec.ts

```diff
@@ -47,7 +50,9 @@ class MockContainer
 	}
 
 	public dispose(error?: ICriticalContainerError): void {
-		this.emit(IFluidContainerSystemEventNames.DISPOSED, error);
+		// IFluidContainer wraps the internal container events and emits its own events that aren't a 1:1 match.
+		// In this case, the "closed" event is instead wrapped on IFluidContainer which will emit "disposed" instead.
+		this.emit("closed", error);
```

Comment:
Since we moved the whole package to work off IFluidContainer, I think it doesn't make sense to mock IContainer anymore, but instead directly mock IFluidContainer and the events from it that we care about (which also avoids the funky discrepancy with dispose/closed). The extra indirection is not gaining us much here, I think, and we can simplify the test code.

Reply:
Couple things,

1. If we care about the mapping of `IContainer`  events to `IFluidContainer` events, this test code ensures the mapping stays as we expect. For example, if `IContainer`'s `close` event no longer maps to `dispose` this test will fail.

2. I have encountered an issue with mocking `IFluidContainer` because `IFluidContainer` gets extra arguments for its own `IFluidContainerEvents` events from its underlying `IContainerEvents` that are hidden from IFluidContainer's interface.
For example, The dispose on `IContainer` is `dispose(error?: ICriticalContainerError): void;`.  and dispose on IFluidContainer is `dispose(): void;` So, I can't mock `IFluidContainer`  with a dispose that I can pass an error object to. Perhaps there is a better way to approach this?

Reply:
1. I don't think unit tests at this layer should care about that mapping. An e2e test might, but the unit test is basically just trying to confirm "If `IFluidContainer` emits event X (whatever it is that caused the event), is this layer generating the appropriate telemetry?".
2. Referring to point 1 and what the unit test wants to validate, I'd say we don't need to mock a `dispose()` call on `IFluidContainer`, we just need the mock object to emit the event we want to test. So we can directly cause the mock to `this.emit("disposed", { someErroObjectWeCraft} )`, right? Which goes back to my original point of "the names of the methods on the mock object don't matter, we just need them to emit the events as we want to test them".

Reply:
Viewed from another perspective, we don't need a mock of a full `IFluidContainer`, we technically just need a mock `IEventProvider<IFluidContainerEvents>`, the "part" of `IFluidContainer` that emits events.

Reply:
Updated.